modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/portlet/FDSViewsPortlet.java

```diff
@@ -427,8 +427,14 @@ private void _createFDSDynamicFilterObjectDefintion(
 						ObjectFieldConstants.BUSINESS_TYPE_TEXT,
 						ObjectFieldConstants.DB_TYPE_CLOB, true, false, null,
 						_language.get(locale, "preselected-values"),
-						"preselectedValues", false)));
+						"preselectedValues", false),
+					ObjectFieldUtil.createObjectField(
+						ObjectFieldConstants.BUSINESS_TYPE_TEXT,
+						ObjectFieldConstants.DB_TYPE_STRING, true, false, null,
+						_language.get(locale, "source"), "source", false)
+					));
```

Comment:
From our discussion yesterday, we should have two new fields:
1. `sourceType`. This one would store strings that are unique identifiers of types: `headless`, `picklist` and later on `vocabulary`.
2. `source`. This one would store value relevant for its type. For `headless`, it will be endpoint URL. For `picklist`, it will be ERC. This field will replace `listTypeDefinitionERC` field.